setup.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+# coding=utf-8
+# flake8: noqa
+# pylint: disable=missing-docstring
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+from setuptools import setup
+
+if __name__ == "__main__":
+    setup(name="funfuzz",
+          version="0.1.0",
+          entry_points={
+              "console_scripts": ["funfuzz = funfuzz.bot:main"]
+          },
+          packages=[
+              "funfuzz",
+              "funfuzz.autobisectjs",
+              "funfuzz.js",
+              "funfuzz.util",
+              "funfuzz.util.tooltool",
+          ],
+          package_data={"funfuzz": [
+              "autobisectjs/*",
+              "js/*",
+              "js/jsfunfuzz/*",
+              "js/shared/*",
+              "util/*",
+              "util/tooltool/*",
+          ]},
+          package_dir={"": "src"},
+          zip_safe=False)
```

Comment:
`requirements.txt` are not pulled in automatically. See other projects like [FuzzManager](https://github.com/MozillaSecurity/FuzzManager/blob/master/setup.py) for one way to do it.

Reply:
Trying to pull in `requirements.txt` automatically is actually quite complex:

* funfuzz requires lithium (which [name-conflicts with an official PyPi package](https://pypi.python.org/pypi/lithium/1.0.1)) and fuzzmanager, both of which are on our GitHub
* [There is a suggestion to hosting your own PyPi server](https://stackoverflow.com/questions/30384322/pip-name-conflict) to allow it to come ahead of the official package
  * The overhead of a separate PyPi server does not seem worth it for now
* `pip install` may be able to properly handle GitHub links when passed in via `-r` directly, and so may `python setup.py install` if `setup.py` is edited, but not via `pip install` **and** `setup.py`. More info [at the comments here](https://stackoverflow.com/a/20234833).
  * `pip` has this [open issue #2124](https://github.com/pypa/pip/issues/2124).
* I tried various combinations of using `subprocess` or even `import pip` in `setup.py` but all seemed to fail for different reasons.

FuzzManager does not have this issue because all its dependencies are on PyPi.

Hence for now, we should pull in the dependencies via a separate `pip install` command.

Reply:
Right, my bad. I knew about this issue. posidron and I had recently been talking about submitting our stuff to pypi, so I went ahead and did [lithium](https://pypi.python.org/pypi/lithium-reducer) but fuzzmanager will take some thinking due to the client/server being separately installable.

Btw there's nothing more or less official about our lithium vs the existing one in pypi, they just beat us to it so we have to rename. I've seen this done before with other pip packages like powerline where the more popular powerline had to use "powerline-status" because of an existing package in pypi.